modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/EditAndScrollTest.java

```diff
@@ -0,0 +1,533 @@
+        ListView<?> control = createAndShowListView(true);
+        fireMouseOnHorizontalTrack(control);
+        assertEquals(editingRow, control.getEditingIndex());
+    }
+
+    @Test
+    public void testListViewFocusedAfterMouseOnVerticalScrollBar() {
+        ListView<?> control = createAndShowListView(false);
+        assertFocusedAfterMouseOnScrollBar(control, Orientation.VERTICAL);
+    }
+
+    @Test
+    public void testListViewFocusedAfterMouseOnHorizontalScrollBar() {
+        ListView<?> control = createAndShowListView(false);
+        assertFocusedAfterMouseOnScrollBar(control, Orientation.HORIZONTAL);
+    }
+
+    /**
+     * Test the test setup for ListView.
+     */
+    @Test
+    public void testListViewEditing() {
+        ListView<?> control = createAndShowListView(true);
+        assertEquals(rows, control.getItems().size());
+        assertEquals(100, scene.getWidth(), 1);
+        assertEquals(330, scene.getHeight(), 1);
+        assertTrue("sanity: vertical scrollbar visible for list " ,
+                getHorizontalScrollBar(control).isVisible());
+        assertTrue("sanity: vertical scrollbar visible for list " ,
+                getVerticalScrollBar(control).isVisible());
+        assertEquals("control must be editing at", editingRow, control.getEditingIndex());
+    }
+
+    /**
+     * Creates and shows a ListView configured to be
+     * - not/editing depending on the startEdit
+     * - scrollable both horizontally and vertically
+     */
+    private ListView<?> createAndShowListView(boolean startEdit) {
+        ObservableList<String> baseData = createData(rows, true);
+        ListView<String> control = new ListView<>(baseData);
+        control.setEditable(true);
+        control.setCellFactory(TextFieldListCell.forListView(new DefaultStringConverter()));
+        showControl(control, true, 100, 330);
+        if (startEdit) {
+            control.edit(editingRow);
+        }
+        return control;
+    }
+
+
+//--------------- test helpers
+
+    /**
+     * Tests that clicking on the scrollbar in the given direction requests
+     * focus back to the given control.
+     *
+     * Note: the control must be showing in the default scene of this test and
+     * have a skin of type VirtualContainerBase.
+     *
+     */
+    private void assertFocusedAfterMouseOnScrollBar(Control control, Orientation dir) {
+        // add and focus additional control
+        Button focusableControl = new Button("dummy");
+        showControl(focusableControl, true);
+        if (dir == Orientation.HORIZONTAL) {
+           fireMouseOnHorizontalTrack(control);
+        } else {
+            fireMouseOnVerticalTrack(control);
+        }
+        assertEquals("virtualized control must be focusOwner after mouse on scrollbar",
+                control, scene.getFocusOwner());
+    }
+
+
+//----------------- Utility methods (TODO: move into infrastructure)
```

Comment:
Can you file a follow-up issue for this?

Reply:
yeah, sure - was undecided whether to do it here or later (locally, there are more that will be added for fixing related issues) - [JDK-8278134](https://bugs.openjdk.java.net/browse/JDK-8278134)